src/parser/mod.rs

```diff
@@ -5275,10 +5275,22 @@ impl<'a> Parser<'a> {
                     };
 
                     let column_def = self.parse_column_def()?;
+
+                    let mut column_position = None;
+                    if dialect_of!(self is MySqlDialect) {
```

Comment:
As per [Contributing Guide](https://github.com/sqlparser-rs/sqlparser-rs?tab=readme-ov-file#new-syntax), this is a dialect specific feature, shold also include `GenericDialect` here?